src/codegen/function.rs

```diff
@@ -283,47 +248,45 @@ pub fn bounds(
 
     let skip_lifetimes = bounds
         .iter()
-        .filter(|bound| skip.contains(&bound.alias))
+        // TODO: False or true?
+        .filter(|bound| bound.alias.map_or(false, |alias| skip.contains(&alias)))
         .filter_map(|bound| match bound.bound_type {
             IsA(Some(lifetime)) | AsRef(Some(lifetime)) => Some(lifetime),
             _ => None,
         })
         .collect::<Vec<_>>();
 
-    let strs: Vec<String> = bounds
+    let lifetimes = bounds
         .iter_lifetimes()
         .filter(|s| !skip_lifetimes.contains(s))
         .map(|s| format!("'{}", s))
-        .chain(
-            bounds
-                .iter()
-                .filter(|bound| {
-                    !skip.contains(&bound.alias)
-                        && (!filter_callback_modified || !bound.callback_modified)
-                })
-                .map(|b| bound_to_string(b, r#async)),
-        )
-        .collect();
-
-    if strs.is_empty() {
-        (String::new(), Vec::new())
+        .collect::<Vec<_>>();
+
+    let bounds = bounds.iter().filter(|bound| {
+        bound.alias.map_or(true, |alias| !skip.contains(&alias))
+            && (!filter_callback_modified || !bound.callback_modified)
+    });
```

Comment:
I need to revisit this: can't be a trait bound if it doesn't have an alias, then it's only written in-place with `impl`.